envoy/event/dispatcher.h

```diff
@@ -228,8 +228,9 @@ class Dispatcher : public DispatcherBase, public ScopeTracker {
    * @return Network::DnsResolverSharedPtr that is owned by the caller.
    */
   virtual Network::DnsResolverSharedPtr
-  createDnsResolver(const std::vector<Network::Address::InstanceConstSharedPtr>& resolvers,
-                    const envoy::config::core::v3::DnsResolverOptions& dns_resolver_options) PURE;
+  createDnsResolver(
+      envoy::config::core::v3::DnsResolutionConfig& dns_resolution_config,
```

Comment:
I don't think we need to lose the `const` in this case, as from an interface point-of-view these configs typically shouldn't be modified.
(Moreover, calling the same function repeatedly with the same config object might have a non-desirable effect because the config has changed).

Reply:
ACK